src/source/Ice/IceAgent.c

```diff
@@ -859,7 +859,7 @@ STATUS iceAgentStateNewTimerCallback(UINT32 timerId, UINT64 currentTime, UINT64
     MUTEX_LOCK(pIceAgent->lock);
     locked = TRUE;
 
-    // prime the state machine
+    // drive the state machine
```

Comment:
Comment for LN 869 as it won't allow me to comment there. I think the best would be to allow the state machine transaction itself to lead to the failed state. So, for example, stepIceAgentStateMachine if failed itself, should result in overall failure. It should still return SUCCESS even if logically moving to an FATAL error state.

Currently, the iceAgentFatalError function simply assigns an error to the status. Perhaps this is the correct action itself.

Reply:
I am gonna address that in a separate PR cuz this one is getting quite lengthy.